src/waveform/visualplayposition.cpp

```diff
@@ -77,7 +83,27 @@ double VisualPlayPosition::getAtNextVSync(VSyncThread* pVSyncThread) {
     if (m_valid) {
         const VisualPlayPositionData data = m_data.getValue();
         const double offset = calcOffsetAtNextVSync(pVSyncThread, data);
-        return data.m_enginePlayPos + offset * data.m_rate;
+
+        double interpolatedPlayPos = data.m_playPos + offset * data.m_playRate;
+
+        if (data.m_loopEnabled) {
```

Comment:
I think this must only applied when the `data.m_playPos` is on one side of the loop marker and the  `interpolatedPlayPos` is on the other side.

Reply:
I've commited a fix.